python/tskit/metadata.py

```diff
@@ -623,8 +632,10 @@ def __init__(self, schema: Optional[Mapping[str, Any]]) -> None:
             codec_instance = codec_cls(schema)
             self._string = tskit.canonical_json(schema)
             self._validate_row = TSKITMetadataSchemaValidator(schema).validate
+            self._bypass_validation = codec_cls.is_schema_trivial(schema)
```

Comment:
Not clear why this is a class method - why not ``codec_instance.is_schema_trivial()``?

Reply:
Just matching the other methods that deal with schemas such as `modify schema`